db/migrate/20141214220050_create_proposals.rb

```diff
@@ -0,0 +1,37 @@
+class CreateProposals < ActiveRecord::Migration
+  def change
+    create_table :proposals do |t|
+      t.string :status
+      t.string :flow
+      t.timestamps
+    end
+    add_reference :carts, :proposal
```

Comment:
For the moment, yes. My plan was to deal with splitting the Cart into use-case-specific models separately.